### PR TITLE
Add list and bye features, UI and exceptions

### PR DESCRIPTION
**Pull request by zoranabc201**
Add list and bye features

### CODE REVIEW

**Comment by ngdeqi** [comment]
Fixes #16
Fixes #13 
Fixes #9

File: src/main/java/seedu/duke/Parser.java [hunk 1]
```diff
@@ -153,6 +168,13 @@ private ArrayList<String> partitionCommandTypeAndDetails(String fullCommandDetai
         return processedCommandDetails;
     }
 
+    void checkForEmptyListParameters(String commandListDetails) throws MissingListDetailException {
```
Comment (wilsonngja) [inline src/main/java/seedu/duke/Parser.java:171]: Would be good to include either public or private. Ambiguous if left empty

File: src/main/java/seedu/duke/Ui.java [hunk 2]
```diff
@@ -263,6 +264,30 @@ public void showEmptyCommandCheckDetailException() {
...
     }
 
+    public void displayOneClient(Client client, int i) {
+        System.out.println(i + ".");
+        System.out.println(client.toString());
+        System.out.println(LINE_BREAK);
+    }
+
+    public void displayOneProperty(Property property, int i) {
+        System.out.println(i + ".");
```
Comment (wilsonngja) [inline src/main/java/seedu/duke/Ui.java:274]: Try to avoid magic literals.

**Review by wilsonngja — COMMENTED** [review]
Try to add more Javadocs header comments.

File: src/main/java/seedu/duke/Parser.java (hunk 1 above)
Comment (OVReader) [reply src/main/java/seedu/duke/Parser.java:171]: private will be good

File: src/main/java/seedu/duke/Parser.java [hunk 3]
```diff
@@ -493,6 +515,16 @@ private int getClientIndexToDelete(String commandDetails) throws EmptyClientInde
         return Integer.parseInt(commandDetails.trim()) - 1;
     }
 
+    private Command prepareForCommandList(String commandDetails) throws IncorrectListDetailsException {
+        if (commandDetails.trim().equals("-client")) {
+            return new CommandListClients();
+        } else if (commandDetails.trim().equals("-property")) {
```
Comment (wilsonngja) [inline src/main/java/seedu/duke/Parser.java:521]: Try to avoid magic literals.

**Pull request merged**